R/check.R

```diff
@@ -1,90 +1,186 @@
+      if (!is.character(value[[i]]) || length(value[[i]]) != 1) {
+        rlang::abort(paste0("argument ", name, "[", i, "] is not a single string"),
+          name = name, value = value, class = "invalid_argument"
+        )
+      }
+    }
+  } else {
     rlang::abort(paste0("argument ", name, " is not a string"),
       name = name, value = value, class = "invalid_argument"
     )
   }
 }
 
+#' Allows a single numeric value
+check_single_int_param <- function(name, value, required = TRUE) {
+  if (is.null(value)) {
+    if (required) {
+      rlang::abort(paste0("argument ", name, " is not a single number"),
+        name = name, value = value, class = "invalid_argument"
+      )
+    }
+  } else if (is.numeric(value)) {
+    if (length(value) != 1) {
+      rlang::abort(paste0("argument ", name, " is not a single number"),
+        name = name, value = value, class = "invalid_argument"
+      )
+    }
+  } else {
+    rlang::abort(paste0("argument ", name, " is not a single number"),
+      name = name, value = value, class = "invalid_argument"
+    )
+  }
+}
+
+#' Allows a vector of numeric values or a list of numeric values
 check_int_param <- function(name, value, required = TRUE) {
-  # numeric or numeric[]
-  if ((required &&
-    !(!is.null(value) &&
-      is.numeric(value))) ||
-    (!required && !(is.null(value) || is.numeric(value)))) {
+  if (is.null(value)) {
+    if (required) {
+      rlang::abort(paste0("argument ", name, " is not a number"),
+        name = name, value = value, class = "invalid_argument"
+      )
+    }
+  } else if (is.numeric(value)) {
+    return()
+  } else if (is.list(value)) {
+    for (i in seq_along(value)) {
+      if (!is.numeric(value[[i]]) || length(value[[i]]) != 1) {
+        rlang::abort(paste0("argument ", name, "[", i, "] is not a single number"),
+          name = name, value = value, class = "invalid_argument"
+        )
+      }
+    }
+  } else {
     rlang::abort(paste0("argument ", name, " is not a number"),
       name = name, value = value, class = "invalid_argument"
     )
   }
 }
 
-is_epirange_like <- function(value) {
-  # character or numeric or EpiRange or list(from=..., to=...)
+#' Return TRUE if value is datelike
+#' TODO: add Date support
+is_date_like <- function(value) {
   is.character(value) ||
-    is.numeric(value) ||
-    inherits(value, "EpiRange") ||
-    (is.list(value) &&
-      "from" %in% names(value) && "to" %in% names(value))
+    is.numeric(value)
 }
 
-check_single_epirange_param <- function(name, value, required = TRUE) {
-  if ((required &&
-    !(!is.null(value) &&
-      is_epirange_like(value))) ||
-    (!required && !(is.null(value) || is_epirange_like(value)))) {
-    rlang::abort(paste0("argument ", name, " is not a epirange"),
+#' Check if param is datelike
+check_single_date_param <- function(name, value, required = TRUE) {
+  if (is.null(value)) {
+    if (required) {
+      rlang::abort(paste0("argument ", name, " is not a datelike"),
```

Comment:
suggestion: "datelike" -> "date_like", in hopes that `??date_like` will do something useful